packages/core/src/browser/shell/theia-dock-panel.ts

```diff
@@ -148,13 +178,21 @@ export class TheiaDockPanel extends DockPanel {
         }
         maximizedElement.style.display = 'block';
         this.addClass(MAXIMIZED_CLASS);
+        console.log(this.preferences);
+        const preference = this.preferences?.get('window.menuBarVisibility') || 'classic';
```

Comment:
It seems that 'classic' is not in use (the if statement refers only to 'visible' value and will ignore undefined value
```suggestion
        const preference = this.preferences?.get('window.menuBarVisibility');
```